src/Instance.ts

```diff
@@ -63,13 +63,13 @@ export class Marked {
           break;
         }
         default: {
-          if (this.defaults.extensions && this.defaults.extensions.childTokens && this.defaults.extensions.childTokens[token.type]) { // Walk any extensions
-            this.defaults.extensions.childTokens[token.type].forEach((childTokens) => {
-              // @ts-expect-error we assume token[childToken] is an array of tokens but we can't be sure
-              values = values.concat(this.walkTokens(token[childTokens], callback));
+          const genericToken = token as Tokens.Generic;
+          if (this.defaults.extensions && this.defaults.extensions.childTokens && this.defaults.extensions.childTokens[genericToken.type]) { // Walk any extensions
+            this.defaults.extensions.childTokens[genericToken.type].forEach((childTokens) => {
```

Comment:
```suggestion
          const array = this.defaults.extensions?.childTokens?.[genericToken.type] ?? []
          // Walk any extensions
          array.forEach((childTokens) => {
```

I think you can simplify this a bit

Reply:
I think we originally had something similar but there was some issue with the optional chaining / nullish coalescing syntax at the time. Is that still the case?

Reply:
We use typescript now with `target` configured, so it should downlevel automatically (if necessary)

Reply:
And I realize now we support node 16 or newer so it already natively supports those features, no downleveling needed

https://node.green